operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/SSABasedGenericKubernetesResourceMatcher.java

```diff
@@ -92,31 +95,26 @@ public boolean matches(R actual, R desired, Context<?> context) {
     var objectMapper = context.getClient().getKubernetesSerialization();
 
     var actualMap = objectMapper.convertValue(actual, Map.class);
-
     sanitizeState(actual, desired, actualMap);
-
-    var desiredMap = objectMapper.convertValue(desired, Map.class);
-    if (LoggingUtils.isNotSensitiveResource(desired)) {
-      log.trace("Original actual: \n {} \n original desired: \n {} ", actual, desiredMap);
-    }
-
     var prunedActual = new HashMap<String, Object>(actualMap.size());
     keepOnlyManagedFields(prunedActual, actualMap,
         managedFieldsEntry.getFieldsV1().getAdditionalProperties(),
         objectMapper);
 
+    var desiredMap = objectMapper.convertValue(desired, Map.class);
```

Comment:
Why was this moved?

Reply:
The original order of method invocations jumped between pruning/sanitizing the actual and the desired resource. Just from the method names it was partially ambiguous to me which resource is changed (actual/desired, actualMap/desiredMap/prunedActual).

So I re-arranged the code into these logical groups:
* get managed fields (or throw)
* get serialization
* prepare actual -> actualMap -> prunedActual
* desired -> desiredMap
* compare prunedActual with desiredMap

By removing some of the many newlines, these logical blocks are also visually grouped together.

Reply:
After going through the code there might be one undesired change: When `keepOnlyManagedFields()` throws an exception, we won't see the `log.trace("Original actual: \n {} \n original desired: \n {} ", actual, desiredMap)` anymore.

If that needs to be fixed, I would suggest to move the whole `desired -> desiredMap` block before the actual block:
```java
    var objectMapper = context.getClient().getKubernetesSerialization();

    var desiredMap = objectMapper.convertValue(desired, Map.class);
    if (LoggingUtils.isNotSensitiveResource(desired)) {
      log.trace("Original actual:\n {}\n original desired:\n {}", actual, desiredMap);
    }
    removeIrrelevantValues(desiredMap);

    var actualMap = objectMapper.convertValue(actual, Map.class);
    sanitizeState(actual, desired, actualMap);
    var prunedActual = new HashMap<String, Object>(actualMap.size());
    keepOnlyManagedFields(prunedActual, actualMap,
        managedFieldsEntry.getFieldsV1().getAdditionalProperties(),
        objectMapper);

    var matches = prunedActual.equals(desiredMap);
```

Reply:
Or we could also align what we print in that trace log:
```java
    var objectMapper = context.getClient().getKubernetesSerialization();
    var actualMap = objectMapper.convertValue(actual, Map.class);
    var desiredMap = objectMapper.convertValue(desired, Map.class);
    if (LoggingUtils.isNotSensitiveResource(desired)) {
      log.trace("Original actual:\n {}\n original desired:\n {}", actualMap, desiredMap);
    }

    sanitizeState(actual, desired, actualMap);
    var prunedActual = new HashMap<String, Object>(actualMap.size());
    keepOnlyManagedFields(prunedActual, actualMap,
        managedFieldsEntry.getFieldsV1().getAdditionalProperties(),
        objectMapper);

    removeIrrelevantValues(desiredMap);

    var matches = prunedActual.equals(desiredMap);
```
I think this is even more readable.

Reply:
Since I anyway had to rebase, I changed the code to the last variant to restore the order of execution, but switched to `actualMap` for the trace logging (I think it'll be much easier to read if both resources are logged in the same format).